model/project_parser.go

```diff
@@ -512,25 +512,43 @@ func (bvt *parserBV) hasSpecificActivation() bool {
 		!utility.FromBoolTPtr(bvt.Activate) || bvt.Disable
 }
 
-// FindAndTranslateProjectForPatch translates a parser project for a patch into a project.
-// This assumes that the version may not exist yet; otherwise FindAndTranslateProjectForVersion is equivalent.
+// FindAndTranslateProjectForPatch translates a parser project for a patch into
+// a project. This assumes that the version may not exist yet (i.e. the patch is
+// unfinalized); otherwise FindAndTranslateProjectForVersion is equivalent.
 func FindAndTranslateProjectForPatch(ctx context.Context, settings *evergreen.Settings, p *patch.Patch) (*Project, *ParserProject, error) {
-	if p.PatchedParserProject == "" {
-		v, err := VersionFindOneId(p.Version)
+	if p.ProjectStorageMethod != "" {
+		pp, err := ParserProjectFindOneByID(ctx, settings, p.ProjectStorageMethod, p.Id.Hex())
 		if err != nil {
-			return nil, nil, errors.Wrapf(err, "finding version '%s' for patch '%s'", p.Version, p.Id.Hex())
+			return nil, nil, errors.Wrapf(err, "finding parser project '%s' in storage", p.Id.Hex())
```

Comment:
Can we add the `ProjectStorageMethod` to the formatted error output?

Reply:
Done.